conans/client/tools/files.py

```diff
@@ -224,17 +225,34 @@ def _manage_text_not_found(search, file_path, strict, function_name, output):
         return False
 
 
+@contextmanager
+def add_permissions(file_path, permissions):
+    if os.path.isfile(file_path):
+        saved_permissions = os.stat(file_path).st_mode
+        if saved_permissions & permissions == permissions:
+            yield
+            return
+        try:
+            os.chmod(file_path, saved_permissions | permissions)
+            yield
+        finally:
+            os.chmod(file_path, saved_permissions)
+    else:
+        yield
+
 def replace_in_file(file_path, search, replace, strict=True, output=None, encoding=None):
     output = default_output(output, 'conans.client.tools.files.replace_in_file')
 
     encoding_in = encoding or "auto"
     encoding_out = encoding or "utf-8"
-    content = load(file_path, encoding=encoding_in)
+    with add_permissions(file_path, stat.S_IREAD):
```

Comment:
Probably the READ case is not a real issue? If some package doesn't have read permission, something will be surely broken? What is a real use case of a package file without READ permission?

Reply:
added mostly for symmetry, but I don't think it happens in reality, so maybe removed

Reply:
Yes, lets remove it, unless until this is reported as a real issue.

Reply:
removed already :)